config/initializers/rack_attack.rb

```diff
@@ -78,6 +81,24 @@ def remote_ip
       end
     end
 
+    # After maxretry OTP requests in findtime minutes,
+    # block all requests from that user for bantime minutes.
+    blocklist('OTP delivery') do |req|
+      # `filter` returns truthy value if request fails, or if it's from a
+      # previously banned user_id so the request is blocked
+      user_id = req.env['warden'].user&.id
+
+      Allow2Ban.filter(
+        "otp-#{user_id}",
+        maxretry: Figaro.env.otp_delivery_blocklist_maxretry.to_i,
+        findtime: Figaro.env.otp_delivery_blocklist_findtime.to_i.minutes,
+        bantime: Figaro.env.otp_delivery_blocklist_bantime.to_i.minutes
```

Comment:
Yes it will since it's at the user level. It doesn't matter where the user tries to send the OTP from. Once they're blocked, they're blocked regardless of their IP.